### ISSUE TITLE
SystemChrome.setPreferredOrientations Failed to change device orientation error. 

### ISSUE THREAD

**Issue by EmirhanSerin1** [flutter/flutter#132127]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

When I try the change device orientation, I have face with this issue. And the current code was working well in old releases.

1. System got [
                          DeviceOrientation.portraitUp,
                        ].
2. Force with [
                          DeviceOrientation.landscapeLeft,
                          DeviceOrientation.landscapeRight,
                        ].
3. When I try to force to landScape it throw the error into console.

And I also have tried another scenerio.

1. System got [
                          DeviceOrientation.portraitUp,
                          DeviceOrientation.landscapeLeft,
                          DeviceOrientation.landscapeRight,
                        ].
2. Force with [
                          DeviceOrientation.landscapeLeft,
                          DeviceOrientation.landscapeRight,
                        ].
3. Force with [
                          DeviceOrientation.portraitUp,
                        ].
4. And same error has occured with different parameter which is "Requested: portraitUp; Supported: landscapeLeft, landscapeRight".

### Expected results

Expected; It supposed to be working well without error as before.

### Actual results

Result; Error

### Code sample

<details><summary>Code sample</summary>

```dart
  onTap: () {
                    if (testValue) {
                      SystemChrome.setPreferredOrientations(
                        [
                          DeviceOrientation.landscapeLeft,
                          DeviceOrientation.landscapeRight,
                        ],
                      );
                    } else {
                      SystemChrome.setPreferredOrientations(
                        [
                          DeviceOrientation.portraitUp,
                        ],
                      );
                    }
                    testValue = !testValue;
                  },
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details><summary>Logs</summary>

```console
Failed to change device orientation: Error Domain=UISceneErrorDomain Code=101 "None of the requested orientations are supported by the view controller. Requested: landscapeLeft, landscapeRight; Supported: portrait" UserInfo={NSLocalizedDescription=None of the requested orientations are supported by the view controller. Requested: landscapeLeft, landscapeRight; Supported: portrait}
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.10.6, on macOS 13.2.1 22D68 darwin-arm64, locale en-TR)
    • Flutter version 3.10.6 on channel stable at /Users/emirhanserin/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f468f3366c (4 weeks ago), 2023-07-12 15:19:05 -0700
    • Engine revision cdbeda788a
    • Dart version 3.0.6
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /Users/emirhanserin/Library/Android/sdk
    • Platform android-33, build-tools 33.0.1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E300c
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] VS Code (version 1.81.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.70.0

[✓] Connected device (3 available)
    • iPhone 14 Pro (mobile) • 116DE120-B7CC-4752-9E6D-123D9B3BA0B1 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 13.2.1 22D68 darwin-arm64
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 115.0.5790.170

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

**Comment by darshankawar** [comment]
@EmirhanSerin1 
Can you take a look at below related issues and check if any of them resembles your case ?

https://github.com/flutter/flutter/issues/34230
https://github.com/flutter/flutter/issues/73651

Also, take a look at [this](https://github.com/flutter/flutter/issues/119473#issuecomment-1415631020) and see if it helps in your case or not.

**Comment by EmirhanSerin1** [comment]
@darshankawar Hello. I have checked these issues and tried the solutions but did not work for me. But as an alternative solution, If I use SystemChrome.setPreferredOrientations([]) before setting orientaion to different DeviceOrientation that is not exists in the orientation list which I set before, It works well. 

Info.plist;
<img width="543" alt="Screenshot 2023-08-10 at 20 50 08" src="https://github.com/flutter/flutter/assets/81233530/620026dd-7995-4cf7-ae77-c6249dc4ea97">
